### PR TITLE
style(editor): Change cursor color in editor in dark mode

### LINKED ISSUE #16284: cursor in the editor is not visible enough in dark mode
#### Issue Description

When the dark mode is enabled, the cursor in the editor's field (coding area) is not visible enough without the higher brightness of the screen. 

#### Browser Information

* Version 63.0.3239.84 (Official Build) (64-bit): 
* macOS High Sierra v10.13.2:  

#### Screenshot
```
https://user-images.githubusercontent.com/23414386/34317416-d2633f26-e7cf-11e7-9064-57b5a7f23289.png
```
P.S. In case you cannot find a cursor in the screenshot, it's in line 9 :)

### PR DESCRIPTION
**Pull request by huyenltnguyen**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #16284

#### Description

The default cursor color is black, which makes it invisible in dark mode. I changed its color (actually `border-color`) to text color to fix it.

### CODE REVIEW

File: common/app/routes/Challenges/views/classic/classic.less [hunk 1]
```diff
@@ -32,6 +32,9 @@
     }
     .cm-keyword, .cm-attribute {
       color:#9BBBDC;
+    }    
+    .CodeMirror-cursor {
```
Comment (Bouncey) [inline common/app/routes/Challenges/views/classic/classic.less]: We need to be more selective here, else `codemirror` styles will win the cascade. 

Use this instead :+1: 
```css
.CodeMirror-cursor {
      border-left-color: @night-text-color;
    }
```

**Review by Bouncey — CHANGES_REQUESTED** [review]

**Comment by huyenltnguyen** [comment]
Changed as requested. I also nested the code inside `.@{ns}-editor .CodeMirror {}` to match the selectors structure.
@Bouncey Thanks for QA'ing my PR.

**Review by DavidCurtis321 — APPROVED** [review]

File: common/app/routes/Challenges/views/classic/classic.less [hunk 2]
```diff
@@ -24,6 +24,9 @@
       background-color:#242424;
```
Comment (tomasvn) [inline common/app/routes/Challenges/views/classic/classic.less]: Add a space after semicolon

Prefer lower case hex values for colors

File: common/app/routes/Challenges/views/classic/classic.less [hunk 3]
```diff
@@ -24,6 +24,9 @@
...
       color:#ABABAB;
```
Comment (tomasvn) [inline common/app/routes/Challenges/views/classic/classic.less]: Add space after semicolon

Prefer lower case hex values for colors

File: common/app/routes/Challenges/views/classic/classic.less (hunk 2 above)
Comment (tomasvn) [inline common/app/routes/Challenges/views/classic/classic.less]: Add a space after semicolon

Prefer lower case hex values for colors

File: common/app/routes/Challenges/views/classic/classic.less [hunk 4]
```diff
@@ -24,6 +24,9 @@
...
     }
+    &-cursor {
+      border-left-color: @night-text-color;
+    }
     .cm-bracket, .cm-tag {
       color:#5CAFD6;
```
Comment (tomasvn) [inline common/app/routes/Challenges/views/classic/classic.less]: Add a space after semicolon

Prefer lower case hex values for colors

**Review by tomasvn — CHANGES_REQUESTED** [review]
~~Fix stylistic inconsistencies across LESS file~~ 

**_Mod update:_** See comment below

**Comment by vkWeb** [comment]
@huyenltnguyen Please make the requested changes. This needs to be merged before shipping. Thanks.

**Comment by raisedadead** [comment]
Hi @tomasvn 

Thanks for the QA,  I agree, you are 100% correct about the best practices. 

But IMHO, OP has not made those (see diff), hence asking them for a change, isn't fair and not even polite.

I am guessing we could take this up as a separate activity, and maybe a good first timers only task?
Would you be interested in creating the issue? or PR?

@vkWeb thanks for the nudge :man_shrugging: , but, see explanation above, same applies.

@huyenltnguyen 
First of all thanks a lot for taking the time to put this together. 

> Edit: I realized that this PR is just papering over the cracks, more details in #16319. I'll leave this PR open as I don't know which solution you guys'd choose, but feel free to close it. Thanks.

Yes, you are absolutely correct.

We have added some additional clarifications in #16319 if you are interested in taking a stab at it.

Meanwhile, I think this is okay to merge.

**Comment by raisedadead** [comment]
Will do a QA, tonight. @Bouncey do you approve of your earlier request for changes?

**Comment by tomasvn** [comment]
@raisedadead I have seen it across different less files, everyone is writing it in a different way, woudn't it be better to have some styleguide? regarding writing consistent CSS? This would help with maintainability of the CSS

**Comment by raisedadead** [comment]
@tomasvn 

> I have seen it across different less files, everyone is writing it in a different way, woudn't it be better to have some styleguide? regarding writing consistent CSS? This would help with maintainability of the CSS

That's an excellent idea.

Here are our current brand guide https://design-style-guide.freecodecamp.org/ but I can understand that we need an dedicated coding style guide.

We can incorporate per your feedback. Would you mind creating an issue on this tracker and add your thoughts / ideas about the CSS style guide?

We are currently updating out docs in general (see tracker for doc issues). Some outline would get us started in adding those as well.

**Comment by huyenltnguyen** [comment]
Thanks @tomasvn and @vkWeb. I noticed the inconsistent code style as well but decided not to touch them as I wasn't sure if this PR was a fix or just a hack (turns out it is the latter).

@raisedadead, thanks for cleaning things up here and also for your explanation in #16319. I think this PR is not necessary as I'm pretty sure we can achieve white cursor when the change discussed in #16319 implemented, and you know what? I made the change on `classic.less`. I haven't tested it yet, but I don't think it works on the `modern` editor (I didn't know we have 2 editors when I submitted this PR). So this PR is ineffective anyway and should be closed. I'm working on the issue right now, have some questions already but I need to do some documenting first. I'll see you in the issue thread and hope you don't mind another wall of text lol.

Thanks again @Bouncey.

**Comment by tomasvn** [comment]
@raisedadead where should I open the issue? here in the fcc repository? or in the appropriate fcc design style guide repo?

**Comment by raisedadead** [comment]
@tomasvn here in this repository, we plan to keep the docs here itself.

**Pull request closed without merging**